### PR TITLE
Feature/analytics (Incorrect name), This branch worked with Demo Checklist

### PR DESCRIPTION
**Pull request by AMadanlal**
- Some of these changes were made when firebase was failing to downloading due to internet speed issues so there will be firebase code that is commented out here and in the development pod.
- Firebase downloaded on the pods but kept failing on the main project when trying to pod install or pod update with the same connection issue mentioned so some firebase stuff will be here.
- made the random game function
- changed more variable names to camelCase

### CODE REVIEW

**Comment by AMadanlal** [comment]
> All good.
> pod side. I saw the commented firebase code that you mentioned. Just make sure to remove it before demo.

I cannot remove it before the demo as I plan to use it when firebase gets installed... eventually

**Comment by XanderSchoeman** [comment]
Alright

File: TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    let platformSearch = utillities.loadMedium()
+    let apiClass = ChickenCoopAPI(searched: "", platform: platformSearch ?? "pc")
+    apiClass.getGameList { result in
+      switch result {
+      case.failure(let error):
+        print(error)
+      case.success(let details):
+        let gameList = details
+        DispatchQueue.main.async {
+          for item in gameList.result where item.platform == platformSearch {
```
Comment (Zamo22) [inline TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift]: You can instead use a filter here directly on details, will look better but not necessary

File: TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    var listItem = GameListItem()
+    var compatableGameList = [GameListItem]()
+    let platformSearch = utillities.loadMedium()
+    let apiClass = ChickenCoopAPI(searched: "", platform: platformSearch ?? "pc")
+    apiClass.getGameList { result in
+      switch result {
+      case.failure(let error):
+        print(error)
+      case.success(let details):
+        let gameList = details
```
Comment (Zamo22) [inline TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift]: No real point naming it details to just name it gameList directly below.
Rather have `let gameList`

File: TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        DispatchQueue.main.async {
+          for item in gameList.result where item.platform == platformSearch {
+              compatableGameList.append(item)
+          }
+          if compatableGameList.count == 1 {
+             listItem = compatableGameList[0]
```
Comment (Zamo22) [inline TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift]: More readable to use compatableGameList.first

File: TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          } else if compatableGameList.count > 1 {
```
Comment (Zamo22) [inline TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift]: Here too, we can check if .isEmpty
Might seem minor but it's good practice to make use of inbuilt functions when we have them and does improve readability

File: TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            let randomInt = Int.random(in: 0..<compatableGameList.count)
```
Comment (Zamo22) [inline TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift]: We can just use `compatableGameList`.randomElement()

File: TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift (hunk 1 above)
Comment (AMadanlal) [reply TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift]: got it

File: TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift (hunk 2 above)
Comment (AMadanlal) [reply TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift]: I just realised that and will definitely change it

File: TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift (hunk 3 above)
Comment (AMadanlal) [reply TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift]: I had no clue a function like that existed... Swift has a lot of tricks up its sleeve.

File: TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift (hunk 4 above)
Comment (AMadanlal) [reply TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift]: I definitely need to learn of the easier way of doing these... swift helps you a lot more than I thought

File: TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift (hunk 5 above)
Comment (AMadanlal) [reply TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift]: I had no clue a function like that actually exists, and I will use it

Comment (AMadanlal) [reply TechReadsPod/CkickenCoopAPI/ViewModel/RandomGameReview.swift]: Thank you for the feedback, I really appreciate all the help you have given us.

**Pull request merged**